mobile/lib/screens/dashboard/dashboard_view.dart

```diff
@@ -464,6 +468,123 @@ class _DashboardViewState extends State<DashboardView>
     }
   }
 
+  Future<void> _sendData() async {
+    AirQualityReading? airQualityReading =
+        context.read<NearbyLocationBloc>().state.locationAirQuality ??
+            HiveService.getAirQualityReadings().firstOrNull;
+    if (airQualityReading == null) return;
+
+    List<ForecastInsight> forecastDbData = await AirQoDatabase()
+        .getForecastInsights(airQualityReading.referenceSite);
+    List<String> forecastValues =
+        forecastDbData.map((e) => e.pm2_5.toString()).take(3).toList();
+    List<String> forecastTimes = List.generate(
+      3,
+      (index) => DateFormat('h a').format(
+        DateTime.now().toLocal().add(
+              Duration(hours: index + 1),
+            ),
+      ),
+    );
+
+    String rectangleWidgetTime =
+        DateFormat('dd/MM, h:mm a').format(DateTime.now().toLocal());
+    String circularWidgetTime =
+        DateFormat('h:mm a').format(DateTime.now().toLocal());
+
+    //TODO: Not working, review later, using manual approach for now
+    // Map<String, String> widgetData = {
+    //   'location': airQualityReading.name,
+    //   'circular_location': airQualityReading.name,
+    //   'date': rectangleWidgetTime,
+    //   'circular_date': circularWidgetTime,
+    //   'pmValue': airQualityReading.pm2_5.toInt().toString(),
+    //   'circular_pm_value': airQualityReading.pm2_5.toInt().toString(),
+    //   'forecastValue1': forecastValues.first,
+    //   'forecastValue2': forecastValues[1],
+    //   'forecastValue3': forecastValues.last,
+    //   'time1': forecastTimes.first,
+    //   'time2': forecastTimes[1],
+    //   'time3': forecastTimes.last,
+    // };
+
+    // return Future.wait([
+    //   widgetData.forEach((key, value) {
+    //     HomeWidget.saveWidgetData<String>(key, value);
+    //   })
+    // ] as Iterable<Future>)
+    //     .then((value) => value);
+
+    return Future.wait([
```

Comment:
Hi @NoahNsimbe , the code below  goes extremely against the DRY principle. I tried to use a forEach loop and map above but it wasn't passing the data as expected thus i opted for what's below, kindly share thoughts

Reply:
We shall review during a peer programming session. We shall create a class for widget data